### PR TITLE
Fix SI-8102 (postfix abs for -0.0)

### PR DESCRIPTION
**Pull request by clhodapp**
SI-8102 points out that -0.0.abs returns -0.0 (in error). The same issue
exists for -0.0f. This commit fixes the issue for both by delegating to
math.abs.

### CODE REVIEW

**Comment by retronym** [comment]
PLS REBUILD ALL

**Comment by retronym** [comment]
Could you please reword the commit title to:

```
SI-8102 postfix abs for -0.0 must return positive 0.0
```

File: test/files/run/SI-8102.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (retronym) [inline test/files/run/SI-8102.scala]: partest test cases should be named `t1234.scala`.

But for tests of the standard library, we are now favouring JUnit tests. This one would belong in `test/files/junit/scala/math/NumericTest.scala` along the lines of `junit/scala/collection/convert/MapWrapperTest.scala`.

**Comment by clhodapp** [comment]
Acknowledged

**Comment by clhodapp** [comment]
Your proposed summary line is too wide (they're supposed to be <= 50 characters wide for some reason), so I stripped the 'positive'. If you'd rather I flout git line width standards, I can throw it back in at your request.

**Comment by som-snytt** [comment]
How about `-0.0.abs must equal 0.0`.  Sorry, sometimes the urge to bike-shed is irresistible.

To me, "postfix abs" sounds like a work-out routine to get rid of a paunchy gut.

**Comment by clhodapp** [comment]
That would be acceptably correct and concise

**Comment by retronym** [comment]
LGTM

File: test/junit/scala/math/NumericTest.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.junit.runners.JUnit4
+
+@RunWith(classOf[JUnit4])
+class NumericTest {
+
+  /* Test for SI-8102 */
+  @Test
+  def testAbs {
+    assertTrue(-0.0.abs equals 0.0)
+    assertTrue(-0.0f.abs equals 0.0f)
```
Comment (retronym) [inline test/junit/scala/math/NumericTest.scala:15]: For next time, `assertEquals(0.0, -0.0f)`

**Pull request merged**